modules/@apostrophecms/soft-redirect/index.js

```diff
@@ -37,7 +37,8 @@ module.exports = {
     return {
       '@apostrophecms/page:notFound': {
         async notFoundRedirect(req) {
-          const urlPathname = req.url.replace(/\?.*$/, '');
+          const url = req._parsedOriginalUrl?.pathname;
+          const urlPathname = url.replace(/\?.*$/, '');
```

Comment:
This regex is redundant since pathname already excludes the ? part of the URL.

Reply:
I wasn't sure. Thanks for the clarification.